### PR TITLE
supports --epub option for opening epub version of docs offline

### LINKED ISSUE #732: Add epub version to documentation
generated from: https://github.com/hexpm/hex/issues/731

current thoughts:

- Generate epub version along side html version and include it in the tarball.
- Add flag for hex.docs tasks for epub (`mix hex.docs offline --epub` would open the epub file instead of default html page.)
- ...

### PR DESCRIPTION
**Pull request by supersimple**
Resolves #732

### CODE REVIEW

**Comment by supersimple** [comment]
@josevalim, et al.
This supports opening epub documentation from hex packages. It assumes the epub doc is named <application-name>.epub
I havent looked into the mix documentation enough to know if that is indeed how the file will always be named.

**Comment by josevalim** [comment]
So the epub is named after the project name, which can be any string. Therefore it is probably safest to do something like `Path.wildcard("*.epub")`and raise relevant error messages if zero or more than one entries are returned.

File: lib/mix/tasks/hex.docs.ex [hunk 1]
```diff
@@ -38,7 +38,7 @@ defmodule Mix.Tasks.Hex.Docs do
   @behaviour Hex.Mix.TaskDescription
 
   @elixir_apps ~w(eex elixir ex_unit iex logger mix)
-  @switches [module: :string, organization: :string, latest: :boolean]
+  @switches [module: :string, organization: :string, latest: :boolean, epub: :boolean]
```
Comment (wojtekmach) [inline lib/mix/tasks/hex.docs.ex]: we need to document this switch.

Comment (wojtekmach) [reply lib/mix/tasks/hex.docs.ex]: Actually, what do you all think about calling this `:format` with the default of `"html'`?

Comment (supersimple) [reply lib/mix/tasks/hex.docs.ex]: the advantage would be flexibility. if a third format was ever available, using the format option would make it easier to support.

**Pull request merged**